flink/src/test/java/org/apache/iceberg/flink/data/RandomData.java

```diff
@@ -88,20 +105,187 @@ public Row next() {
     };
   }
 
+  private static Iterable<Record> generateIcebergGenerics(Schema schema, int numRecords,
```

Comment:
Seems it could share the common code with RandomGenericData#generate ?  Make the `RandomGenericData#generate` to return a Iterable<Record> ?

Reply:
You are right, let me refactor this.

Reply:
This method accepts a `Record` supplier and then generate records. We should keep it for generating fallback records and dictionary encoded records.  But for `generateRecords` method we can update it to call `RandomGenericData#generate` directly.